File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/nomis/repository/ReferenceCodeRepository.java
package uk.gov.justice.digital.hmpps.oauth2server.nomis.repository;

import org.springframework.data.repository.CrudRepository;
import uk.gov.justice.digital.hmpps.oauth2server.nomis.model.DomainCodeIdentifier;
import uk.gov.justice.digital.hmpps.oauth2server.nomis.model.ReferenceCode;
import uk.gov.justice.digital.hmpps.oauth2server.nomis.model.ReferenceDomain;

import java.util.List;

public interface ReferenceCodeRepository extends CrudRepository<ReferenceCode, DomainCodeIdentifier> {

    List<ReferenceCode> findByDomainCodeIdentifierDomainAndActiveIsTrueAndExpiredDateIsNull(ReferenceDomain domain);
}

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/nomis/repository/ReferenceCodeRepository.kt
package uk.gov.justice.digital.hmpps.oauth2server.nomis.repository

import org.springframework.data.repository.CrudRepository
import uk.gov.justice.digital.hmpps.oauth2server.nomis.model.DomainCodeIdentifier
import uk.gov.justice.digital.hmpps.oauth2server.nomis.model.ReferenceCode
import uk.gov.justice.digital.hmpps.oauth2server.nomis.model.ReferenceDomain

interface ReferenceCodeRepository : CrudRepository<ReferenceCode, DomainCodeIdentifier> {
  fun findByDomainCodeIdentifierDomainAndActiveIsTrueAndExpiredDateIsNull(domain: ReferenceDomain?): List<ReferenceCode>
}
